website/docs/api-layers.md

```diff
@@ -1409,15 +1409,44 @@ https://github.com/explosion/thinc/blob/master/thinc/layers/with_array.py
 Flatten nested inputs on the way into a layer and reverse the transformation
 over the outputs.
 
-| Argument    | Type                                                             | Description        |
-| ----------- | ---------------------------------------------------------------- | ------------------ |
-| `layer`     | <tt>Model[Sequence[Sequence[Any]], Sequence[Sequence[Any]]]</tt> | The layer to wrap. |
-| **RETURNS** | <tt>Model[ListXd, ListXd]</tt>                                   | The wrapped layer. |
+<infobox variant="warning">
+
+Even though `with_flatten` is a layer wrapper, it does not preserve symmetry
+between the input and output data types. This often makes it hard to compose
+with other layers. [`with_flatten_v2`](#with_flatten_v2) instead.
+
+</infobox>
+
+| Argument    | Type                                            | Description        |
+| ----------- | ----------------------------------------------- | ------------------ |
+| `layer`     | <tt>Model[Sequence[Any], ArrayXd]</tt>          | The layer to wrap. |
+| **RETURNS** | <tt>Model[Sequence[Sequence[Any]], ListXd]</tt> | The wrapped layer. |
 
 ```python
 https://github.com/explosion/thinc/blob/master/thinc/layers/with_flatten.py
 ```
 
+### with_flatten_v2 {#with_flatten_v2 tag="function" new="8.1.6"}
```

Comment:
I mean `with_flatten_v2` here is weird as the reference. The registered `with_flatten.v2` is fine/normal, but do these descriptions in the docs use the `_v2` names?

Reply:
Outside the scope of this PR, but wouldn’t we want downstream users to get these functions through the registry, because then we can move these functions more freely (like into a future ‘spacy-legacy’).

Reply:
We do want them to use the registry, so that's part of why I think it's weird that the API docs are presented like this. The registry names aren't even included anywhere.

Reply:
Yeah, I'll make a ticket for this.